cart-service/src/main/java/com/petclinic/cartsservice/businesslayer/CartServiceImpl.java

```diff
@@ -39,7 +39,17 @@ public Mono<CartResponseModel> getCartByCartId(String cartId) {
                             .stream().map(productId -> productClient.getProductByProductId(productId).flux())
                             .reduce(Flux.empty(), Flux::merge)
                             .collectList()
-                            .map(products -> EntityModelUtil.toCartResponseModel(cart, products));
+                            .map(products -> {
+                                // Calculate subtotal, tvq, tvc, and total
+                                double subtotal = products.stream()
+                                        .mapToDouble(product -> product.getProductSalePrice() * (product.getQuantity() != null ? product.getQuantity() : 1))
+                                        .sum();
+                                double tvq = subtotal * 0.09975; // 9.975%
```

Comment:
What do you think of making the variables tvq and tvc as final variables in order to make them constants as they are fixed rates?

Reply:
They are not final as they are dependent on subtotal, so they are recalculated dynamically.